securedrop/source_app/main.py

```diff
@@ -108,7 +108,19 @@ def create() -> werkzeug.Response:
             if not date_codenames_expire or datetime.now(timezone.utc) >= date_codenames_expire:
                 return clear_session_and_redirect_to_logged_out_page(flask_session=session)
 
-            tab_id = request.form["tab_id"]
+            tab_id = request.form.get("tab_id")
+            if not tab_id or tab_id not in session.get("codenames", {}):
+                # Use generic error message text for source creation issue
+                flash_msg(
```

Comment:
I don't see this message appearing, I believe because we're immediately clearing the session afterwards. If I move the session.clear() above the flash_msg, then it works.

Would be good to have the test also verify the message is flashed if that's straightforward to do.